src/generators/generator.py

```diff
@@ -0,0 +1,6 @@
+from abc import ABC, abstractmethod
+
+class AbstractGenerator(ABC):
```

Comment:
For better readability, put all generators in one file (similar to parser.py), i.e.

```python
class GeneratorType(Enum):
    INT = 'int'
    FLOAT = 'float'
    STRING = 'String'
    UUID = 'UUID'
    DATE = 'date'
    PERIOD = 'period'
    
class AbstractGenerator(ABC):
    [...]

class FloatGenerator(Generator):
    [...]

class StringGenerator(Generator):
    [...]

class GeneratorFactory:
  def create_generator(self, generator_type: str, value_set: str, constraints: Dict[str, Any]) -> Generator:
    params = {}
    if generator_type == GeneratorType.INT:
      return self._create_int_generator(params)
    elif generator_type == GeneratorType.FLOAT:
      return self._create_float_generator(params)
    else:
      raise ValueError(f"Unknown generator type: {generator_type}")

  def _create_int_generator() -> IntGenerator:
    [...]

  def _create_float_generator()> -> FloatGenerator:
  [...]

```